src/cli/build_command.zig

```diff
@@ -190,18 +190,18 @@ pub const BuildCommand = struct {
         try this_transpiler.configureDefines();
         this_transpiler.configureLinker();
 
-        if (ctx.bundler_options.production) {
-            bun.assert(!this_transpiler.options.jsx.development);
-        }
-
         if (!this_transpiler.options.production) {
             try this_transpiler.options.conditions.appendSlice(&.{"development"});
         }
 
         this_transpiler.resolver.opts = this_transpiler.options;
         this_transpiler.resolver.env_loader = this_transpiler.env;
-        this_transpiler.options.jsx.development = !this_transpiler.options.production;
-        this_transpiler.resolver.opts.jsx.development = this_transpiler.options.jsx.development;
+
+        // Allow tsconfig.json overriding, but always set it to false if --production is passed.
+        if (ctx.bundler_options.production) {
+            this_transpiler.options.jsx.development = false;
+            this_transpiler.resolver.opts.jsx.development = false;
+        }
```

Comment:
This change is a no-op. And this PR does not move the assertion later as claimed in the description, it just deletes it entirely.